### PR TITLE
Refactor test suite

### PR DESCRIPTION
**Pull request by cskaandorp**
Test suite is drastically refactored. Tests are now executed independently, and all projects-api tests are now bundled into a single file in which a parametrised fixture runs an auth and non-auth version of the app. In the folder structure I have made room for integration tests and extension test(s) (forthcoming). Documentation is also forthcoming.
Last but not least: there was a collaboration icon still present in the code. That has been taken care of.
This PR will also solve Issue #1383

### CODE REVIEW

**Comment by PeterLombaers** [comment]
Closes #1383

File: asreview/webapp/tests/conftest.py [hunk 1]
```diff
@@ -20,96 +20,85 @@
...
-def setup_teardown_signed_in():
-    """Setup and teardown with a signed in user."""
-    # setup environment variables
-    os.environ.update(TMP_ENV_VARS)
-    # load appropriate config file
-    root_dir = str(Path(os.path.abspath(__file__)).parent)
-    config_file_path = f"{root_dir}/configs/auth_config.json"
-    # create app and client
-    app = create_app(enable_auth=True, flask_configfile=config_file_path)
+ASREVIEW_PATH=str(Path("~", ".asreview-test").expanduser())
```
[the remark this answers is not in the document]
Comment (cskaandorp) [reply asreview/webapp/tests/conftest.py]: Yes indeedy!

File: asreview/webapp/tests/data/asreview-project-v0-19-startreview/project.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"version": "0.19", "id": "asreview-project-v0-19-startreview", "name": "ASReview Project v0.19 startReview", "description": "", "authors": "", "created_at_unix": 1648717742, "projectInitReady": true, "reviewFinished": false, "dataset_path": "Kwok_2020.csv"}
```
Comment (J535D165) [inline asreview/webapp/tests/data/asreview-project-v0-19-startreview/project.json:1]: Is this a replacement for the use of `asreview/asreview-project-files-testing`? I'm not a big fan of having large files in the repo (I try to remove them). Would a git submodule be useful?

**Review by J535D165 — CHANGES_REQUESTED** [review]
Looks very interesting. Only one remark on the test project, see comment.

File: asreview/webapp/tests/data/asreview-project-v0-19-startreview/project.json (hunk 2 above)
Comment (cskaandorp) [reply asreview/webapp/tests/data/asreview-project-v0-19-startreview/project.json:1]: Yeah I think so, lemme have a look.

Comment (cskaandorp) [reply asreview/webapp/tests/data/asreview-project-v0-19-startreview/project.json]: Fixed with downloading projects from Github into the asreview test folder. Open for enhancement.

**Review by J535D165 — APPROVED** [review]

**Comment by J535D165** [comment]
Very nice refactor!! Thanks to @terrymyc for this work on this in the past. The start of something nice.

**Pull request merged**